install-dependencies.sh

```diff
@@ -80,73 +81,140 @@ if [ -z "$DISTRO" ]; then
    su urpmi gcc gcc-c++ binutils make asio-devel SDL_image-devel SDL_ttf-devel \
     SDL_mixer-devel png-devel gettext-devel cmake SDL_gfx-devel jpeg-devel \
     tiff-devel git glew-devel
+
 elif [ "$DISTRO" == "debian" ]; then
    echo "Installing dependencies for Debian/Ubuntu Linux, Linux Mint..."
    sudo apt install git cmake g++ gcc gettext libasio-dev libglew-dev libpng-dev libsdl2-dev \
     libsdl2-image-dev libsdl2-mixer-dev libsdl2-ttf-dev python zlib1g-dev
+
 elif [ "$DISTRO" == "freebsd" ]; then
    echo "Installing dependencies for FreeBSD..."
    pkg install git asio cmake gettext glew png sdl2_image sdl2_mixer sdl2_net sdl2_ttf
+
 elif [ "$DISTRO" == "openbsd" ]; then
    echo "Installing dependencies for OpenBSD..."
-   pkg_add git cmake gcc g++ gettext-tools glew icu4c libexecinfo png \
+   doas pkg_add git cmake gcc g++ gettext-tools glew icu4c libexecinfo png \
     sdl2-image sdl2-mixer sdl2-net sdl2-ttf
-   echo "Asio is not packaged for OpenBSD, please install it from source:"
-   echo " - Download the latest stable version from"
-   echo "      https://sourceforge.net/projects/asio/files/asio/"
-   echo " - Unpack in your development directory"
-   echo " - Run './configure --prefix=/usr && make install-data' in the unpacked asio"
-   echo "   directory (or just copy recursively 'asio.hpp' and the 'asio' subdirectory"
-   echo "   from 'include' to '/usr/include')"
+   if ! [ -f /usr/include/asio.hpp -o -f ./auto_dependencies/asio/asio.hpp ]
+   then
+      asio_not_packaged "OpenBSD" "doas"
+      exit 1
+   fi
+
 elif [ "$DISTRO" == "msys32" ]; then
    echo "Installing dependencies for 32-bit Windows..."
    pacman -S pacman -S mingw-w64-i686-toolchain git mingw-w64-i686-cmake \
     mingw-w64-i686-ninja mingw-w64-i686-asio mingw-w64-i686-SDL2_ttf \
     mingw-w64-i686-SDL2_mixer mingw-w64-i686-SDL2_image \
     mingw-w64-i686-glbinding mingw-w64-i686-glew mingw-w64-i686-icu
+
 elif [ "$DISTRO" == "msys64" ]; then
    echo "Installing dependencies for 64-bit Windows..."
    pacman -S mingw-w64-x86_64-toolchain git mingw-w64-x86_64-cmake \
     mingw-w64-x86_64-ninja mingw-w64-x86_64-asio mingw-w64-x86_64-SDL2_ttf \
     mingw-w64-x86_64-SDL2_mixer mingw-w64-x86_64-SDL2_image \
     mingw-w64-x86_64-glbinding mingw-w64-x86_64-glew mingw-w64-i686-icu
+
 elif [ "$DISTRO" == "homebrew" ]; then
    echo "Installing dependencies for Mac Homebrew..."
    brew install asio git cmake doxygen gettext glew graphviz icu4c jpeg \
     libogg libpng libvorbis ninja python sdl2 sdl2_image sdl2_mixer sdl2_ttf zlib
+
+elif [ "$DISTRO" == "solus" ]; then
+   echo "Installing dependencies for Solus..."
+   sudo eopkg install -c system.devel
+   sudo eopkg install git gettext glew-devel libicu-devel libpng-devel sdl2-devel \
+    sdl2-image-devel sdl2-mixer-devel sdl2-ttf-devel python
+
+   if ! [ -f /usr/include/asio.hpp -o -f ./auto_dependencies/asio/asio.hpp ]
+   then
+      echo "Installing Asio from source..."
```

Comment:
This looks like it might also work on OpenBSD, so it should be inside a separate function.

Reply:
I was thinking about it too, but it's not quite that simple, and none of us seems to be familiar with OpenBSD. Some web searching shows that there are some incompatibilities that need to be handled:
- neither `wget`, nor `curl` is installed by default on OpenBSD, but they have an improved `ftp` command that can work non-interactively and can actually do HTTP as well
- they have `md5` instead of `md5sum`, and it has a different output format. It needs `md5 -r` for the `cksum` / GNU `md5sum` format
- BSD `tar` is different from GNU `tar`. Reading their [man page](https://man.openbsd.org/tar), I think `-j` is mandatory (I wasn't even aware that GNU `tar` doesn't need it any more for reading archives until I looked it up because of this change) -- but at least `tar -xjf` probably works the same with both